Makefile

```diff
@@ -0,0 +1,95 @@
+ifeq ($(GOPATH),)
+export GOPATH=/tmp/go
+endif
+export PATH := $(PATH):$(GOPATH)/bin
+
+INSTALL := /usr/bin/install
+DBDIR := /var/run/redis/sonic-db/
+GO ?= /usr/local/go/bin/go
+TOP_DIR := $(abspath ..)
+BUILD_DIR := build/bin
+export GOBIN := $(abspath $(BUILD_DIR))
+export PATH := $(PATH):$(GOBIN):$(shell dirname $(GO))
+
+SRC_FILES=$(shell find . -name '*.go' | grep -v '_test.go' | grep -v '/tests/')
+BLD_FLAGS := -tags readwrite
+
+GO_DEPS := vendor/.done
+PATCHES := $(wildcard patches/*.patch)
+UNIT_TEST := $(shell if sudo [ -e /var/run/redis/redis.sock ]; then echo "exist"; else echo "noexist"; fi)
+
+all: sonic-gnmi
+
+go.mod:
+	$(GO) mod init github.com/sonic-net/sonic-gnmi
+
+$(GO_DEPS): go.mod $(PATCHES)
+	$(GO) mod vendor
+	$(GO) mod download golang.org/x/crypto@v0.0.0-20191206172530-e9b2fee46413
+	$(GO) mod download github.com/jipanyang/gnxi@v0.0.0-20181221084354-f0a90cca6fd0
+	cp -r $(GOPATH)/pkg/mod/golang.org/x/crypto@v0.0.0-20191206172530-e9b2fee46413/* vendor/golang.org/x/crypto/
+	cp -r $(GOPATH)/pkg/mod/github.com/jipanyang/gnxi@v0.0.0-20181221084354-f0a90cca6fd0/* vendor/github.com/jipanyang/gnxi/
+	bash patches/apply.sh vendor
+	chmod -R u+w vendor
+	patch -d vendor -p0 < patches/gnmi_cli.all.patch
+	patch -d vendor -p0 < patches/gnmi_set.patch
+	patch -d vendor -p0 < patches/gnmi_get.patch
+	patch -d vendor -p0 < patches/path.patch
+	touch $@
+
+go-deps: $(GO_DEPS)
+
+go-deps-clean:
+	$(RM) -r vendor
+
+sonic-gnmi: $(GO_DEPS)
+ifeq ($(CROSS_BUILD_ENVIRON),y)
+	$(GO) build -o ${GOBIN}/telemetry -mod=vendor $(BLD_FLAGS) github.com/sonic-net/sonic-gnmi/telemetry
+	$(GO) build -o ${GOBIN}/gnmi_get -mod=vendor github.com/jipanyang/gnxi/gnmi_get
+	$(GO) build -o ${GOBIN}/gnmi_set -mod=vendor github.com/jipanyang/gnxi/gnmi_set
+	$(GO) build -o ${GOBIN}/gnmi_cli -mod=vendor github.com/openconfig/gnmi/cmd/gnmi_cli
+	$(GO) build -o ${GOBIN}/gnoi_client -mod=vendor github.com/sonic-net/sonic-gnmi/gnoi_client
+	$(GO) build -o ${GOBIN}/gnmi_dump -mod=vendor github.com/sonic-net/sonic-gnmi/gnmi_dump
+else
+	$(GO) install -mod=vendor $(BLD_FLAGS) github.com/sonic-net/sonic-gnmi/telemetry
+	$(GO) install -mod=vendor github.com/jipanyang/gnxi/gnmi_get
+	$(GO) install -mod=vendor github.com/jipanyang/gnxi/gnmi_set
+	$(GO) install -mod=vendor github.com/openconfig/gnmi/cmd/gnmi_cli
+	$(GO) install -mod=vendor github.com/sonic-net/sonic-gnmi/gnoi_client
+	$(GO) install -mod=vendor github.com/sonic-net/sonic-gnmi/gnmi_dump
+endif
+
+check:
+ifeq ("$(UNIT_TEST)", "exist")
+ifeq ($(wildcard ${DBDIR}/database_config.json),)
+	sudo mkdir -p ${DBDIR}
+	sudo cp ./testdata/database_config.json ${DBDIR}
+endif
+	cd ./gnmi_server && sudo $(GO) test -coverprofile=coverage-gnmi.txt -covermode=atomic -mod=vendor -coverpkg ./,../sonic_data_client
```

Comment:
>./,../sonic_data_client [](http://example.com/codeflow?start=109&length=23)

Limit coverage in some folders will not make sense in long run because we care about PR diff coverage, and it may hide some lines in hidden folders.

We could override this checker for the initial commits in this PR.